src/SmartComponents/redux/reducers.js

```diff
@@ -18,3 +18,33 @@ export function composeReducer(state = { }, action) {
             return state;
     }
 }
+
+const initialPendingComposeState = {
+    release: {
+        arch: 'x86_64',
+        release: 'rhel-8',
+    },
+    uploadDestinations: {
+        aws: false,
+        azure: false,
+        google: false,
+    },
+};
+
+export function pendingCompose(state = initialPendingComposeState, action) {
+    let newState = Object.assign({}, state);
+    switch (action.type) {
+        case types.SET_RELEASE:
+            newState.release = { arch: action.arch, release: action.release };
+            return newState;
+        case types.SET_UPLOAD_DESTINATION:
+            newState.uploadDestinations = {
+                aws: action.aws,
+                azure: action.azure,
+                google: action.google,
```

Comment:
I personally am a fan of using spread operators. Any thoughts on using them through out the state?
 
```
switch (action.type) {
  case types.SET_RELEASE:
    return {
      ...state,
      release: {
        arch: action.arch,
        release: action.release 
      }
    };
```